nnc/Store.swift

```diff
@@ -133,10 +133,63 @@ private let fpzipDecode:
     return 1
   }
 
-#if canImport(Compression)
-  import Compression
+private let zipEncode:
+  @convention(c) (
+    UnsafeRawPointer?, Int, Int32, UnsafePointer<Int32>?, Int32, UnsafeMutableRawPointer?,
+    UnsafeMutableRawPointer?, UnsafeMutablePointer<Int>?, UnsafeMutablePointer<UInt32>?
+  ) -> Int32 = {
+    data, dataSize, dataType, dimensions, dimensionCount, context, encoded, encodedSize,
+    identifier
+    in
+    guard let data = data, let dimensions = dimensions, let encoded = encoded,
+          let encodedSize = encodedSize, dimensionCount > 0
+    else { return 0 }
+    var zippedSize = encodedSize[0]
+    guard zip(data: data, dataSize: dataSize, zippedData: encoded, zippedDataSize: &zippedSize) else {
+      return 0
+    }
+    identifier?[0] = 0x217
+    encodedSize[0] = zippedSize
+    return 1
+  }
+
+private let zipDecode:
+  @convention(c) (
+    UnsafeRawPointer?, Int, Int32, UnsafePointer<Int32>?, Int32, UInt32, UnsafeMutableRawPointer?,
+    UnsafeMutableRawPointer?, UnsafeMutablePointer<Int>?
+  ) -> Int32 = {
+    data, dataSize, dataType, dimensions, dimensionCount, identifier, context, decoded,
+    decodedSize
+    in
+    guard identifier == 0x217 else { return 0 }
+    guard let data = data, let dimensions = dimensions, let decoded = decoded,
+          let decodedSize = decodedSize, dimensionCount > 0
+    else { return 0 }
+    var unzippedDataSize = decodedSize[0]
+    guard unzip(data: data, dataSize: dataSize, unzippedData: decoded, unzippedDataSize: &unzippedDataSize) else { return 0 }
+    decodedSize[0] = unzippedDataSize
```

Comment:
is this what it's supposed to be? i wasn't sure of the exact meaning of `decodedSize[0] = decodedSize[0] - stream.dst_size` in the previous unzipping code

Reply:
Previously we decode until we reached the end (hence - stream_dst_size) or we decoded enough to fill up all the bytes we offered (in case you only read partial tensor).

I believe this is the case you didn't address hence the confusion. See: https://github.com/liuliu/s4nnc/blob/main/test/store.swift#L255